lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/driver/RuntimesHost.java

```diff
@@ -0,0 +1,190 @@
+      }
+
+      runtimeInjector = runtimeInjector.forkInjector(config);
+      this.runtimesInjectors.put(rd.getRuntimeName().toString(), runtimeInjector);
+    }
+  }
+
+  private void initializeInjector(final Injector runtimeInjector) throws InjectionException {
+    EventHandler<ResourceStatusEvent> statusEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.ResourceStatusHandler.class);
+    runtimeInjector.bindVolatileParameter(RuntimeParameters.ResourceStatusHandler.class, statusEventHandler);
+    EventHandler<NodeDescriptorEvent> nodeDescriptorEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.NodeDescriptorHandler.class);
+    runtimeInjector.bindVolatileParameter(RuntimeParameters.NodeDescriptorHandler.class, nodeDescriptorEventHandler);
+    EventHandler<ResourceAllocationEvent> resourceAllocationEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.ResourceAllocationHandler.class);
+    runtimeInjector.bindVolatileParameter(
+            RuntimeParameters.ResourceAllocationHandler.class,
+            resourceAllocationEventHandler);
+    EventHandler<RuntimeStatusEvent> runtimeStatusEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.RuntimeStatusHandler.class);
+    runtimeInjector.bindVolatileParameter(
+            RuntimeParameters.RuntimeStatusHandler.class,
+            runtimeStatusEventHandler);
+  }
+
+  private RuntimeDefinition parseSerializedRuntimeDefinition(final String serializedRuntimeDefinition) throws
+          IOException {
+    RuntimeDefinition rd;
+    final JsonDecoder decoder = DecoderFactory.get().
+            jsonDecoder(RuntimeDefinition.getClassSchema(), serializedRuntimeDefinition);
+    final SpecificDatumReader<RuntimeDefinition> reader = new SpecificDatumReader<>(RuntimeDefinition.class);
+    rd = reader.read(null, decoder);
+    return rd;
+  }
+
+  private Injector getInjector(final String requestedRuntimeName) {
+    String runtimeName = requestedRuntimeName;
+    if (runtimeName.isEmpty()) {
+      runtimeName = defaultRuntimeName;
+    }
+
+    return this.runtimesInjectors.get(runtimeName);
+  }
+
+  public void onNext(final ResourceLaunchEvent value) {
+    try {
+      System.err.println("Resource launch called for runtime " + value.getRuntimeName());
+      getInjector(value.getRuntimeName()).getInstance(ResourceLaunchHandler.class).onNext(value);
+    } catch (InjectionException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public void onNext(final RuntimeStart value) {
+    System.err.println("Runtime start called for runtime ");
+    initialize();
+    try {
+      for (Injector runtimeInjector : this.runtimesInjectors.values()) {
+        runtimeInjector.getInstance(ResourceManagerStartHandler.class).onNext(value);
+      }
+    } catch (InjectionException e) {
+      e.printStackTrace();
```

Comment:
Please don't swallow exceptions, here and elsewhere. Also, we have gone through some work to make sure that exceptions in Java get logged exactly once.

Also, it would be better to catch the exception at each level.